elastic/synthetics
# fix: filter monitors by tags added through monitor.use

Fixes: https://github.com/elastic/synthetics/issues/775

# Context

the `--tags` flag in the push command was only taking into account tags passed as an argument in the `journey` function

```
journey({ name: 'My Example Journey', tags: ['foo']}, ({ page, params }) => {
   // ... 
})
```

# Fix

Tags added through `monitor.use` are now included in the journey before filtering monitors by `tag`.


## How to test

1. Create [project monitor](https://www.elastic.co/guide/en/observability/current/synthetics-get-started-project.html#synthetics-get-started-project-init)
2. Make sure you add the tag via `monitor.use`.  For instance:

  ```
journey('My Example Journey', ({ page, params }) => {
  monitor.use({
    tags: ['foo']
  });

  step('launch application', async () => {
    await page.goto(params.url);
  });

  step('assert title', async () => {
    const header = await page.locator('h1');
    expect(await header.textContent()).toBe('todos');
  });
});

  ```
 
3. execute `SYNTHETICS_API_KEY=YOUR_KEY npm run push`. You should see 4 added tests (at least if you stay with the default tests).
4. execute `SYNTHETICS_API_KEY=YOUR_KEY npm run push -- --tags foo`. You should see 3 removed tests and one unchanged

**Recording**:


https://github.com/elastic/synthetics/assets/15065076/7f1d3f60-0658-4257-9cee-1580a185522a
